src/anndata/_io/specs/methods.py

```diff
@@ -110,7 +121,9 @@ def wrapper(
 @_REGISTRY.register_read(H5File, IOSpec("", ""))
 @_REGISTRY.register_read(H5Group, IOSpec("", ""))
 @_REGISTRY.register_read(H5Array, IOSpec("", ""))
-def read_basic(elem, _reader):
+def read_basic(
+    elem: H5File | H5Group | H5Array, *, _reader: Reader
```

Comment:
I think `h5py.File` is a subclass of `h5py.Group` so technically there’s no need to specify it in any read signature, but we can just leave it in of course.